src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/TabOrder.cs

```diff
@@ -245,31 +211,36 @@ private void DrawTabs(IList tabs, Graphics gr, bool fRegion)
                 }
                 else
                 {
-                    Control p = ctlHover.Parent;
-                    Color backColor;
-                    backColor = p.BackColor;
-
-                    Region clip = gr.Clip;
-                    gr.ExcludeClip(ctlInner);
-                    using (SolidBrush brush = new SolidBrush(backColor))
+                    if (ctlHover.Parent is Control hoverParent)
                     {
-                        gr.FillRectangle(brush, ctlOuter);
-                    }
+                        Color backColor;
+                        backColor = hoverParent.BackColor;
+
+                        Region clip = gr.Clip;
+                        gr.ExcludeClip(ctlInner);
+                        using (SolidBrush brush = new SolidBrush(backColor))
+                        {
+                            gr.FillRectangle(brush, ctlOuter);
+                        }
 
-                    ControlPaint.DrawSelectionFrame(gr, false, ctlOuter, ctlInner, backColor);
-                    gr.Clip = clip;
+                        ControlPaint.DrawSelectionFrame(gr, false, ctlOuter, ctlInner, backColor);
+                        gr.Clip = clip;
+                    }
                 }
             }
 
+            int iCtl = 0;
+            string str;
             Rectangle rc;
-            while (e.MoveNext())
+            Control? parent;
+            foreach (Control ctl in tabs)
```

Comment:
nit: we are avoiding using abbreviated names.
```suggestion
            foreach (Control control in tabs)
```